core/jazz_is-slack-channel-available/index.js

```diff
@@ -0,0 +1,163 @@
+        } else {
+          return cb(JSON.stringify(errorHandler.throwInternalServerError("Unhandled error.")));
+        }
+      });
+  } catch (e) {
+    logger.error(e);
+    return cb(JSON.stringify(errorHandler.throwInternalServerError("Unhandled error.")));
+  }
+};
+
+function genericInputValidation(event) {
+  logger.debug("inside genericInputValidation");
+  return new Promise((resolve, reject) => {
+    if (!event || !event.method) {
+      reject({
+        result: "inputError",
+        message: "method cannot be empty"
+      });
+    }
+
+    if (event.method !== 'GET') {
+      reject({
+        result: "inputError",
+        message: "Unsupported method/request"
+      });
+    }
+
+    if (event.method === 'GET' && (!event.query || !Object.keys(event.query).includes('slack_channel') || !event.query.slack_channel)) {
+      reject({
+        result: "inputError",
+        message: "Missing input parameter slack_channel"
+      });
+    }
+
+    if (!event.principalId) {
+      reject({
+        result: "unauthorized",
+        message: "Unauthorized."
+      });
+    }
+
+    resolve(event.query.slack_channel);
+  });
+}
+
+function requestToChannels(config, resObj, channel_name) {
+  logger.debug("Inside requestToChannels:" + channel_name);
+  return new Promise((resolve, reject) => {
+    var slack_token = "token=" + config.slack_channel_token;
+    //Pull endpoints from config file
+    var public_channel_url = config.public_channel_endpoint + slack_token;
+    var priv_channel_url = config.priv_channel_endpoint + slack_token;
+    var urlList = [];
+    urlList.push(getResponse(public_channel_url, channel_name));
+    urlList.push(getResponse(priv_channel_url, channel_name));
+
+    Promise.all(urlList)
+      .then(res => {
+        if (res.includes('true')) {
+          resObj.is_available = true;
+        } else {
+          resObj.is_available = false;
+        }
+        resolve(resObj);
+      })
+      .catch(error => {
+        logger.error("error: " + JSON.stringify(error));
+        reject(error);
+      });
+  });
+}
+
+function getResponse(channel_url, channel_name) {
+  logger.debug("Inside getResponse:" + channel_url + ",channel_name:" + channel_name);
+  return new Promise((resolve, reject) => {
+    var params = {
+      method: 'GET',
+      uri: channel_url,
+      rejectUnauthorized: false
+    };
+
+    request(params, (error, response, body) => {
+      if (error) {
+        logger.error(error);
+        reject(error);
+      } else {
+        var data = JSON.parse(response.body);
+        if (data.ok) {
+          var list = data.channels ? data.channels : data.groups;
+          if (list.length) {
+            var count = 0;
+            list.find(each =>{
+              if(each.name === channel_name) {
+                resolve('true');
+              } else {
+                count++;
+                if (count === list.length) {
+                  resolve('false');
```

Comment:
are you sure this works. if the first one in the loop goes to else then it will just resolve to false and return. I dont think this will ever resolve to true. Have you tested this?

Reply:
Yes it works. 
If the first one in the loop goes to else then it will check for the count value with list length based on that check it resolves with false and return. If count value is !== list.length, then it take the next one in the loop.